File: language-code/doc-kotlin/Getting StartedCoding Conventions_19_f5ca166894e9fe829fcf1b69484658e6.kt
abstract class Foo<out T : Any> : IFoo {
    abstract fun foo(a: Int): T
}
​
class FooImpl : Foo() {
    constructor(x: String) : this(x) { /*...*/ }
    
    val x = object : IFoo { /*...*/ } 
}

File: language-code/doc-kotlin/Getting StartedCoding Conventions_19_f5ca166894e9fe829fcf1b69484658e6.swift
protocol IFoo {}
class AImpl: IFoo { /*...*/ }

protocol Foo: IFoo {
    associatedtype T
    func foo(a: Int) -> T
}

class FooImpl: Foo {
    init(x: String) { /*...*/ }
    
    func foo(a: Int) -> String { /*...*/ }
    
    let x: IFoo = AImpl()
}
